addons/website_quote/models/order.py

```diff
@@ -213,37 +188,41 @@ def onchange_template_id(self, cr, uid, ids, template_id, partner=False, fiscal_
         date = False
         if quote_template.number_of_days > 0:
             date = (datetime.datetime.now() + datetime.timedelta(quote_template.number_of_days)).strftime("%Y-%m-%d")
-        data = {'order_line': lines, 'website_description': quote_template.website_description, 'note': quote_template.note, 'options': options, 'validity_date': date}
-        return {'value': data}
 
-    def recommended_products(self, cr, uid, ids, context=None):
-        order_line = self.browse(cr, uid, ids[0], context=context).order_line
-        product_pool = self.pool.get('product.product')
+        self.order_line = lines
+        self.website_description = quote_template.website_description
+        self.note = quote_template
+        self.options = options
+        self.validity_date = date
+
+    @api.multi
+    def recommended_products(self):
         products = []
-        for line in order_line:
-            products += line.product_id.product_tmpl_id.recommended_products(context=context)
+        for line in self.order_line:
+            products += line.product_id.product_tmpl_id.recommended_products()
         return products
 
-    def get_access_action(self, cr, uid, id, context=None):
+    @api.multi
+    def get_access_action(self):
```

Comment:
api.one() return list.so instead of using like self[0].i use api.multi.is it correct?